src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeControllerTest.php

```diff
@@ -1782,6 +1782,29 @@ public function testCopyWithShadow()
         $this->assertStringStartsWith('/test_en/test_en', $englishDocument->getResourceSegment());
     }
 
+    public function testUnpublish()
+    {
+        $document = $this->createPageDocument();
+        $document->setTitle('test_de');
+        $document->setParent($this->homeDocument);
+        $document->setStructureType('default');
+        $this->documentManager->persist($document, 'de');
+        $this->documentManager->publish($document, 'de');
+        $this->documentManager->flush();
+
+        $client = $this->createAuthenticatedClient();
+
+        $client->request(
+            'POST',
+            '/api/nodes/' . $document->getUuid() . '?action=unpublish&language=de'
+        );
+
+        $this->assertHttpStatusCode(200, $client->getResponse());
```

Comment:
Also add check if the published date and workflowstage have been reset to null and test.